Build fails on CentOS 5

Build fails on CentOS 5 (gcc-4.1.2, glibc-2.5):

```
$ make
cc -Wall -Werror -D_GNU_SOURCE -g   -c -o reptyr.o reptyr.c
In file included from platform/linux/linux.h:36,
                 from platform/platform.h:26,
                 from reptyr.c:37:
/usr/include/sys/wait.h:104: error: expected identifier before numeric constant

$ sed -ne '100,108p' /usr/include/sys/wait.h
/* The following values are used by the `waitid' function.  */
#if defined __USE_SVID || defined __USE_XOPEN
typedef enum
{
  P_ALL,		/* Wait for any child.  */
  P_PID,		/* Wait for specified process.  */
  P_PGID		/* Wait for members of process group.  */
} idtype_t;
#endif
```

--- a/platform/linux/linux.h
+++ b/platform/linux/linux.h
@@ -24,6 +24,14 @@
 #define LINUX_H
 
 #ifdef __linux__
+
+// Important that we include <sys/wait.h> before <linux/*>, for
+// compatibility with certain older libcs, in which <linux/wait.h>
+// #define's some symbols that <sys/wait.h> defines via
+// `enum. c.f. https://bugs.launchpad.net/ubuntu/+source/eglibc/+bug/1261872/
+// and https://github.com/nelhage/reptyr/issues/67
+#include <sys/wait.h>
+
 #include <linux/major.h>
 #include <linux/net.h>
 #include <linux/limits.h>
@@ -33,7 +41,6 @@
 #include <asm/ptrace.h>
 #include <sys/types.h>
 #include <sys/user.h>
-#include <sys/wait.h>
 #include <unistd.h>
 #include <stdlib.h>
 #include <stdio.h>

